track_remote.go

```diff
@@ -125,12 +127,23 @@ func (t *TrackRemote) Read(b []byte) (n int, attributes interceptor.Attributes,
 		}
 	}
 
-	n, attributes, err = r.readRTP(b, t)
-	if err != nil {
-		return
+	// If there's a separate RTX track and an RTX packet is available, return that
+	if rtxPacket, rtxAttributes := r.readRTX(t); rtxPacket != nil {
+		n, err = rtxPacket.MarshalTo(b)
```

Comment:
It may not be worth the optimization without benchmarks, but I wonder if it is worth somehow saving the marshaled packet (maybe in attributes?), since `ReadRTP` will have to unmarshal immediately anyway. I checked some profiles from our deployment, and it looks like `Unmarshal()` is only ~2.8% of the CPU time used in `ReadRTP()` for us, and `MarshalTo()` is less than 1% of `WriteRTP()`.

Reply:
I thought about this too, but I don't think this is worth it. If we store the marshalled packet (so we can reuse it in `ReadRTP`) we are just trading off saving a tiny amount of CPU for a tiny amount of extra memory usage. The CPU saving would only be when processing RTX packets, which are typically only 5% of all received packets anyway. So by doing this we would be looking at saving 5% of 2.8% ~= 0.14% in Read + ReadRTP.